force-app/main/default/classes/AccountAutoDeletionSettingsVMapper.cls

```diff
@@ -0,0 +1,124 @@
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @date 2021
+ * @group
+ * @group-content ../../ApexDocContent/EdaSettings.htm
+ * @description View mapper to generate view models associated with Account Auto Deletion funtionality.
+ */
+public virtual with sharing class AccountAutoDeletionSettingsVMapper {
+    /**********************************************************************************
+     * @description Instance for Singleton Pattern
+     **********************************************************************************/
+    @TestVisible
+    private static AccountAutoDeletionSettingsVMapper instance;
+
+    /**********************************************************************************
+     * @description Empty constructor is not public for Singleton Pattern
+     **********************************************************************************/
+    protected AccountAutoDeletionSettingsVMapper() {
+    }
+
+    /**********************************************************************************
+     * @description Static method to get the current instance for Singleton pattern
+     * @return The instance of AccountAutoDeletionSettingsVMapper.
+     **********************************************************************************/
+    public static AccountAutoDeletionSettingsVMapper getInstance() {
+        if (instance == null) {
+            instance = new AccountAutoDeletionSettingsVMapper();
+        }
+
+        return instance;
+    }
+
+    /*******************************************************************************************************
+     * @description Retrieve a DualListboxVModel representing the Account Record Types currently selected
+     * for deletion when not associated with Contact records and any remaining active account record types
+     * availabe for selection.
+     * @return  A DualListboxVModel
+     *******************************************************************************************************/
+    public virtual DualListboxVModel getAccountAutoDeletionSettingsViewModel() {
+        // get current values set for deletion from hierarchy settings
+        AccountRecordTypeAutoDeletionModel currentAutoDeletionSettings = this.locateAccountRecordTypeAutoDeletionService()
+            .getAccountRecordTypeAutoDeletionModel();
+
+        // get all active account record types in org
+        Map<Id, RecordTypeModel> activeAccountRecordTypeModelsById = this.locateRecordTypeService()
+            .getActiveRecordTypeModelsById(
+                this.locateRecordTypeService().getModelByIdMapForSObjectType(Account.SObjectType).values()
```

Comment:
This is efficient. We tend to assign local variables here, but this is also refactor-safe. I'm happy keeping this as it is.